src/main/java/org/gridsuite/explore/server/ExploreService.java

```diff
@@ -0,0 +1,176 @@
+                                    )
+            );
+        });
+    }
+
+    public Mono<Void> replaceFilterContingencyListWithScript(UUID id, String userId) {
+        return directoryService.getElementInfos(id).flatMap(elementAttributes -> {
+            if (!userId.equals(elementAttributes.getOwner())) {
+                return Mono.error(new ExploreException(NOT_ALLOWED));
+            }
+            return contingencyListService.replaceFilterContingencyListWithScript(id);
+        });
+    }
+
+    public Mono<Void> createFilter(String filter, String filterName, Boolean isPrivate, UUID parentDirectoryUuid, String userId) {
+        ElementAttributes elementAttributes = new ElementAttributes(null, filterName, FILTER,
+                new AccessRightsAttributes(isPrivate), userId, 0);
+
+        return directoryService.createElement(elementAttributes, parentDirectoryUuid, userId).flatMap(elementAttributes1 ->
+                        filterService.insertFilter(filter, elementAttributes1.getElementUuid(), userId)
+                                .doOnError(err ->
+                                    directoryService.deleteElement(elementAttributes1.getElementUuid(), userId).subscribe()
+                                )
+        );
+    }
+
+    public Mono<Void> newScriptFromFilter(UUID filterId, String scriptName, String userId, UUID parentDirectoryUuid) {
+        return directoryService.getElementInfos(filterId).flatMap(elementAttributes -> {
+            if (!elementAttributes.getType().equals(FILTER)) {
+                return Mono.error(new ExploreException(NOT_ALLOWED));
+            }
+            ElementAttributes newElementAttributes = new ElementAttributes(null, scriptName,
+                    FILTER, new AccessRightsAttributes(elementAttributes.getAccessRights().isPrivate()), userId, 0);
+            return directoryService.createElement(newElementAttributes, parentDirectoryUuid,  userId).flatMap(elementAttributes1 ->
+                filterService.insertNewScriptFromFilter(filterId, scriptName, elementAttributes1.getElementUuid())
+                    .doOnError(err ->
+                        directoryService.deleteElement(elementAttributes1.getElementUuid(), userId).subscribe()
+                    )
+            );
+        });
+    }
+
+    public Mono<Void> replaceFilterWithScript(UUID id, String userId) {
+        return directoryService.getElementInfos(id).flatMap(elementAttributes -> {
+            if (!userId.equals(elementAttributes.getOwner())) {
+                return Mono.error(new ExploreException(NOT_ALLOWED));
+            }
+            if (!elementAttributes.getType().equals(FILTER)) {
+                return Mono.error(new ExploreException(NOT_ALLOWED));
+            }
+            return filterService.replaceFilterWithScript(id)
+                    .doOnSuccess(unused ->
+                        directoryService.updateElementType(id, FILTER, userId)
```

Comment:
directoryService shouldn't update type if It doesn't know the subtypes. Maybe it's only a TODO, not sure if the PR is still WIP or not...

Reply:
Removed